src/components/common/button/index.tsx

```diff
@@ -14,20 +15,36 @@ export type ButtonHTMLProps = DetailedHTMLProps<
 > &
   AriaAttributes;
 
-export type ButtonProps = ButtonHTMLProps & VariantProps<typeof buttonStyles>;
+export type ButtonProps = ButtonHTMLProps &
+  VariantProps<typeof buttonStyles> & {
+    loading?: boolean;
+    loadingChildren?: string | ReactNode;
+  };
 
 export const Button: FC<ButtonProps> = ({
   variant,
   size,
   fullWidth,
   className,
+  loading,
+  loadingChildren,
   ...props
 }) => {
   return (
     // @ts-ignore
     <m.button
       className={buttonStyles({ variant, size, fullWidth, class: className })}
       {...props}
-    />
+      disabled={props.disabled || loading}
+    >
+      {loading ? (
+        <>
+          {loadingChildren || props.children}
+          <span className="dot-pulse ml-30 flex" />
```

Comment:
why does this element need flex class?

Reply:
in order to make the spaces between the dots animation evenly - if I remove the flex property, we will get the distorted animation like we had before.

Reply:
is this specific to only here or why not add it to the dot-pulse class? Wont this happen again when we use the animation in another place?

Reply:
Yea you right. moving flex styling to index.css